src/EppoClient.php

```diff
@@ -360,16 +375,179 @@ private function getAssignmentDetail(
+     *
+     * Actions are selected based on the subject and action contexts. Contexts are passed as associative arrays
+     * (key=>value) of attributes which, by default, are sorted into numeric and non-numeric. Non-numeric attributes
+     * are bucketed as "Categorical Attributes" while numeric are treated as "Numeric Attributes". Demonstrated in the
+     * **Example 2** below is a means of explicitly classifying attributes as either Numeric or Categorical.
+     *
+     * Example 1:
+     *
+     * $flagKey = 'my-bandit-flag';
+     * $subject = 'user-123';
+     * $subjectContext = ['accountAge' => 0.5, 'country' => 'US'];
+     *
+     * // A simple list of actions with no context attributes
+     * $actions = ['nike', 'adidas', 'reebok'];
+     *
+     * $result = $client->getBanditAction($flagKey, $subject, $subjectContext, $actions, 'control');
+     *
+     * Example 1.2: Actions with un-grouped attributes
+     * $actions = [
+     *   'nike': [
+     *     'brandLoyalty' => 0.0,
+     *     'size' => 5,
+     *     'colour' => 'red'
+     *   ], ...
+     * ];
+     *
+     * $result = $client->getBanditAction($flagKey, $subject, $subjectContext, $actions, 'control');
+     *
+     * Example 2:
+     *
+     * $subjectContext = new AttributeSet(
+     *      numericAttributes: ['accountAge' => 0.5],
+     *      categoricalAttributes: ['zip' => 90210, 'country' => 'US']
+     * );
+     *
+     * $actions = [
+     *   'nike': new AttributeSet(
+     *     numericAttributes: ['brandLoyalty' => 0.0],
+     *     categoricalAttributes: ['size' => 5, 'colour' => 'red']
+     *   ), ...
+     * ];
+     *
+     * $result = $client->getBanditAction($flagKey, $subject, $subjectContext, $actions, 'control');
+     *
+     * @param string $flagKey
+     * @param string $subjectKey
+     * @param array<string, ?object>|AttributeSet $subjectContext
+     * @param array<string>|array<string, array<string, ?object>>|array<string, AttributeSet> $actions
+     * @param string $defaultVariation
+     * @return BanditResult
+     *
+     * @throws EppoClientException
+     */
+    public function getBanditAction(
+        string $flagKey,
+        string $subjectKey,
+        array|AttributeSet $subjectContext,
+        array $actions,
+        string $defaultVariation
+    ): BanditResult {
+        try {
+            // Normalize the subject and action into AttributeSets. These functions detect the structure of the
+            // user data allowing for either automatic or manual sorting into numeric and categorical attributes.
+            $subject = AttributeSet::fromFlexibleInput($subjectContext);
+            $actionContexts = AttributeSet::arrayFromFlexibleInput($actions);
+
+            return $this->getBanditDetail($flagKey, $subjectKey, $subject, $actionContexts, $defaultVariation);
+        } catch (EppoException $e) {
+            // TODO: handle the different exception types.
```

Comment:
what do you mean by this?

Reply:
I was thinking that there might be some smart handling that could be done with exceptions encountered. After spending more time with it, I don't think there's anything to do.